Automattic-Tracks-iOS/Crash Logging/CrashLogging.swift

```diff
@@ -2,178 +2,251 @@ import Foundation
 import Sentry
 import CocoaLumberjack
 
+struct CrashLoggingInternals {
+    static var crashLogging: CrashLogging?
+}
+
 /// A class that provides support for logging crashes. Not compatible with Objective-C.
 public class CrashLogging {
 
-    /// A singleton is maintained, but the host application needn't be aware of its existence.
-    internal static let sharedInstance = CrashLogging()
-
-    private var dataProvider: CrashLoggingDataProvider?
-    private var eventLogging: EventLogging?
+    private let dataProvider: CrashLoggingDataProvider
+    private let eventLogging: EventLogging?
 
-    /// Thread-safe single initialization
-    fileprivate static let threadSafeDispatchQueue = DispatchQueue(label: Bundle.main.bundleIdentifier ?? "tracks" + "-crash-logging-queue")
-    fileprivate static var _isStarted = false
-    internal static var isStarted: Bool {
-        get {
-            return threadSafeDispatchQueue.sync { _isStarted }
-        }
-        set {
-            threadSafeDispatchQueue.sync { _isStarted = newValue }
-        }
+    public init(dataProvider: CrashLoggingDataProvider, eventLogging: EventLogging? = nil) {
+        self.dataProvider = dataProvider
+        self.eventLogging = eventLogging
     }
 
-    /**
-     Initializes the crash logging system.
-
-     - Parameters:
-     - dataProvider: An object that will provide any required data to the crash logging system.
-
-     - SeeAlso: CrashLoggingDataProvider
-     */
-    public static func start(withDataProvider dataProvider: CrashLoggingDataProvider, eventLogging: EventLogging? = nil) {
+    public func start() throws -> CrashLogging {
 
-        // Only allow initializing this system once
-        guard !isStarted else { return }
-        isStarted = true
+        /// Validate the DSN ourselves before initializing, because the SentrySDK silently prints the error to the log instead of telling us if the DSN is valid
```

Comment:
```suggestion
        // Validate the DSN ourselves before initializing, because the SentrySDK silently prints the error to the log instead of telling us if the DSN is valid
```

I'm suggesting to go from `///` to `//` because the former is to be used for documentation comments, but this one is not an header doc of a type or method.

Reply:
> the Sentry SDK silently prints the error to the log instead of telling us if the DSN is valid

Have you considered opening an issue with a feature suggestion for this.

Reply:
Yeah, I like the internals thing a lot – done in 673c92240be83c52fcbd3768a99b45117f1304d7

Reply:
Regarding the `///` vs `//` thing – WooCommerce started the trend of using the `///` format for comments because they support markdown, links, etc (and are far smaller).

There's some of that already in the ExPlat stuff, so I've been aiming to use the `///` format everywhere. However, I did have instances of `//` that _should_ have been annotated with `MARK:`, so I've added that where appropriate.

Reply:
Okay. Maybe I'm wrong about Apple's desired purpose for `///` vs `//`.

Anyways, it's just a style thing 👌

Reply:
> Maybe I'm wrong about Apple's desired purpose

Probably not – we just abuse it for other things 😅